### PR TITLE
Enabling github test workflows for all branches.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
Signed-off-by: AWSHurneyt <hurneyt@amazon.com>

### Description
Enabling github actions when submitting PRs directly to version branches.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
The cypress test workflow is failing because it needs to build against 2.2 if we're using OSD branch `2.x`. We bumped our `2.x` branch to `2.2` but didn't do it to `main` since we were planning on bumping it to `3.0` anyway. The bump to `3.0` is currently blocked on a bug being tracked in the OpenSearch repo as mentioned in https://github.com/opensearch-project/notifications/issues/481

If the bump to `3.0` remains blocked, we'll get `main` on to `2.2` in the meantime or update the CI to properly build against OSD `2.1`. The former would be the more consistent option since it would be strange to have our `2.x` branch ahead of `main` in version.

**Comment by AWSHurneyt** [comment]
The `link checker` action is failing due to this link in the `README`. It looks like other plugins are still referencing this webpage in their `README` files, but the link was broken when the opendistro repos were deprecated.
```
https://discuss.opendistrocommunity.dev/c/Use-this-category-for-all-questions-around-machine-learning-plugins

**Review by lezzago — DISMISSED** [review]

File: .github/workflows/cypress-workflow.yml [hunk 1]
```diff
@@ -3,10 +3,14 @@ on:
   pull_request:
     branches:
       - main
+      - 1.x
```
Comment (qreshi) [inline .github/workflows/cypress-workflow.yml:6]: Am I correct in assuming this means we don't want this to run on `1.3` and before?

Comment (AWSHurneyt) [reply .github/workflows/cypress-workflow.yml:6]: That's a good point. I based this configuration on the one used by the alerting dashboards workflows. We may want to reconfiguring both repos to check the `1.*` branches considering the new branching strategy. For now, would it make sense to merge this as-is, and discuss enabling the checks for a future PR?

**Comment by AWSHurneyt** [comment]
> The `link checker` action is failing due to this link in the `README`. It looks like other plugins are still referencing this webpage in their `README` files, but the link was broken when the opendistro repos were deprecated.
> 
> ```
> https://discuss.opendistrocommunity.dev/c/Use-this-category-for-all-questions-around-machine-learning-plugins
> ```

Discussed this with @bowenlan-amzn , who recommended changing the link to this one https://forum.opensearch.org/c/plugins/index-management. I'll revise this PR.

**Comment by AWSHurneyt** [comment]
Closing in favor of https://github.com/opensearch-project/index-management-dashboards-plugin/pull/224/files

**Pull request closed without merging**